edb/server/protocol/auth_ext/base.py

```diff
@@ -37,14 +43,98 @@ def __init__(
         self.client_secret = client_secret
         self.http_factory = http_factory
 
-    def get_code_url(self, state: str, redirect_uri: str) -> str:
+    async def get_code_url(self, state: str, redirect_uri: str) -> str:
         raise NotImplementedError
 
-    async def exchange_code(self, code: str) -> str:
+    async def exchange_code(self, code: str) -> data.OAuthAccessTokenResponse:
         raise NotImplementedError
 
-    async def fetch_user_info(self, token: str) -> data.UserInfo:
+    async def fetch_user_info(
+        self, token_response: data.OAuthAccessTokenResponse
+    ) -> data.UserInfo:
         raise NotImplementedError
 
     def _maybe_isoformat_to_timestamp(self, value: str | None) -> float | None:
         return datetime.fromisoformat(value).timestamp() if value else None
+
+
+class OpenIDProvider(BaseProvider):
+    def __init__(self, name: str, issuer_url: str, *args, **kwargs):
+        super().__init__(name, issuer_url, *args, **kwargs)
+
+    async def get_code_url(self, state: str, redirect_uri: str) -> str:
+        oidc_config = await self._get_oidc_config()
```

Comment:
My OIDC knowledge is a bit rusty here - are we supposed to cache the OIDC config for some time?

Reply:
Ah, nvm, now I see you're using `httpx_cache` below.